### DISCUSSION TITLE
How to check whether an existing trajectory is in collision?

### DISCUSSION THREAD

**Discussion by pmj110119** [NVlabs/curobo#102] [Software Q&A] [answered]
Hi guys, thanks for your great work.

I wonder how to check whether an existing trajectory is in collision?

I generated a longer trajectory for the robot via `MotionGeneration`. However, as the robot moves along the trajectory, the environment changes. I want to manually check whether this path collides with the current environment, and stop running if it collides.

I tried looking through the curobo source code, but didn't find it. Is there a simple method similar to `check_collisiton(plan_state)` that can be called directly to determine collision?

**Comment by balakumar-s** [comment, marked as answer]
Check this example: https://curobo.org/source/getting_started/2a_python_examples.html#collision-checking
